web/src/components/merchant/sign-up/SignUpCard.tsx

```diff
@@ -56,31 +65,123 @@ const StyledButton = styled(Button)`
   font-size: 18px;
   font-weight: bolder;
   text-transform: uppercase;
-
-  margin-top: 30px;
+  margin: 0;
 `;
 
 /**
  * This is the card containing the sign up form for the Merchant app.
  * It is displayed in the Sign Up page.
  */
-const SignUpCard: React.FC = () => (
-  <CardContainer>
-    <GroupBuyMerchantHeader />
-    <Form>
-      <FormRow label="Name" inputType="text" />
-      <FormRow label="Email" inputType="email" />
-      <FormRow label="Password" inputType="password" />
-      <FormRow label="Confirm Password" inputType="password" />
-      <FormRow label="VPA" inputType="email" />
+const SignUpCard: React.FC = () => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState('');
+
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const [vpa, setVpa] = useState('');
+  const [vpaError, setVpaError] = useState('');
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
prefer to move this above to put together with name variables. probably a separated function?

Reply:
What do you mean by a separated function? The reason why I put this here is because it's a function and I want the above to show only the states, all together.

Reply:
we'd better to group by logic instead of state then function.